### PR TITLE
Add New Attributes

### PR DESCRIPTION
**Pull request by johncblandii**
## what
* Added additional performance tuning parameters
* Added basic example

## why
* New enhancements available from underlying terraform resource

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -1,3 +1,8 @@
+output "arn" {
+  value       = "${aws_efs_file_system.default.arn}"
+  description = "EFS arn"
```
Comment (osterman) [inline outputs.tf]:
```suggestion
  description = "EFS ARN"
```

File: outputs.tf [hunk 2]
```diff
@@ -1,3 +1,8 @@
...
+}
+
 output "id" {
   value       = "${aws_efs_file_system.default.id}"
   description = "EFS id"
```
Comment (osterman) [inline outputs.tf]:
```suggestion
  description = "EFS ID"
```

File: example/main.tf [hunk 3]
```diff
@@ -0,0 +1,23 @@
+provider "aws" {
+  region = "${var.region}"
+}
+
+data "aws_availability_zones" "available" {}
```
Comment (aknysh) [inline example/main.tf]: we don't need this since we have the var `var.availability_zones`

Comment (johncblandii) [reply example/main.tf]: Copy/pasta from my local where I didn't have the var.

I'll fix it up later.

File: outputs.tf [hunk 4]
```diff
@@ -22,3 +32,8 @@ output "mount_target_ips" {
   value       = ["${aws_efs_mount_target.default.*.ip_address}"]
   description = "List of IPs of the EFS mount targets (one per Availability Zone)"
 }
+
+output "network_interface_ids" {
+  value       = ["${aws_efs_mount_target.default.*.network_interface_id}"]
+  description = "The ID of the network interface that Amazon EFS created when it created the mount target."
```
Comment (aknysh) [inline outputs.tf]: The IDs of the network interfaces

File: example/variables.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+variable "name" {
+  default = "efs"
+}
+
+variable "stage" {
+  default = "testing"
+}
+
+variable "availability_zones" {}
```
Comment (aknysh) [inline example/variables.tf]: Please update to this format to make it consistent:
variable "availability_zones" {
   type = "list"
}

Add `type = "list"` for all list types .

File: example/variables.tf [hunk 6]
```diff
@@ -0,0 +1,27 @@
+variable "region" {
+  default = "us-east-1"
+}
+
+variable "attributes" {
+  default = []
```
Comment (aknysh) [inline example/variables.tf]: Add `type = "map"

Comment (osterman) [reply example/variables.tf]: type `list`, no?

Comment (johncblandii) [reply example/variables.tf]: `attributes`:

```
variable "attributes" {
  type        = "list"
  default     = []
  description = "Additional attributes (e.g. `policy` or `role`)"
}
```

**Comment by johncblandii** [comment]
Ready for a final review.

File: example/variables.tf (hunk 6 above)
Comment (aknysh) [reply example/variables.tf]: ah sorry, `attributes` have type `list` (was thinking about `tags`)

File: outputs.tf [hunk 7]
```diff
@@ -9,10 +14,15 @@ output "host" {
 }
 
 output "dns_name" {
-  value       = "${aws_efs_file_system.default.id}.efs.${var.aws_region}.amazonaws.com"
+  value       = "${aws_efs_file_system.default.dns_name}"
   description = "DNS name"
 }
 
+output "mount_target_dns_names" {
+  value       = "${aws_efs_mount_target.default.*.dns_name}"
```
Comment (aknysh) [inline outputs.tf]:
```suggestion
  value       = "[${aws_efs_mount_target.default.*.dns_name}"]
```

Comment (aknysh) [reply outputs.tf]: @johncblandii  sorry, I missed this.
Since we use `[]` in all other outputs (e.g. `network_interface_ids`), let's make it consistent.

Comment (johncblandii) [reply outputs.tf]: Done

**Pull request merged**